polyply/src/topology.py

```diff
@@ -332,9 +355,22 @@ def gen_bonded_interactions(self):
                             atoms = " ".join(list(map(lambda x: str(x), interaction.atoms)))
                             raise OSError(msg.format(inter_type, atoms))
 
-                        interaction.parameters[:] = new_params[:]
-                        if meta:
-                            interaction.meta.update(meta)
+                        for idx, (new_param, meta) in enumerate(new_params):
+                            if not meta:
+                                meta = {}
+                            if idx > 0:
+                                new_interaction = Interaction(atoms=tuple(interaction.atoms),
+                                                              parameters=new_param,
+                                                              meta=meta)
+                                additional_interactions[inter_type].append(new_interaction)
+                            else:
+                                interaction.parameters[:] = new_param[:]
+                                interaction.meta.update(meta)
```

Comment:
I've read this code wrong a few times, wrote a wrong suggestion, and am still a bit confused. 
Could you swap the if and else statements? And add some comments?